hw/ip/otbn/dv/otbnsim/sim/wsr.py

```diff
@@ -86,33 +86,53 @@ def changes(self) -> List[TraceWSR]:
 
 
 class RandWSR(WSR):
-    '''The magic RND WSR'''
+    '''The magic RND WSR
+
+    RND is special as OTBN can stall on reads to it. A read from RND either
+    immediately returns data from a cache of a previous EDN request (triggered
+    by writing to the RND_PREFETCH CSR) or waits for data from the EDN. To model
+    this anything reading from RND must first call `request_value` which returns
+    True if the value is available.
+    '''
     def __init__(self, name: str):
         super().__init__(name)
 
-        # For now, the RTL doesn't have a real "random number generator".
-        # Eventually, it will have an LFSR of some sort, seeded by the
-        # CSRNG/EDN. We'll model that properly when we've specced it out. Until
-        # then, random numbers are constant.  This constant must match the one
-        # in the RTL (the `rnd` signal in the `otbn_core` module found in
-        # rtl/otbn_core.sv). If changed here it must be changed there to match.
-        # Constant for RND is the binary bit pattern 1001 (0x9 hex) repeated to
-        # fill a 256-bit word.
-        u32 = 0x99999999
-        u64 = (u32 << 32) | u32
-        u128 = (u64 << 64) | u64
-        self._random_value = (u128 << 128) | u128
+        self._random_value = None
+        self._random_value_read = False
+        self.requesting_rnd = False
```

Comment:
I'm not sure about the name here. Maybe "pending_request" or something? Otherwise, you get a repeat of the "rnd" part of the name.